From e3732fa378cce958f1c51bbc9608b13a144ca7f8 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Thu, 6 Jul 2023 12:48:42 +0200
Subject: [PATCH 316/954] fix(stm32mp2): syscfg: Add delay block stop API

This patch adds a new delay block service (stop the delay block).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 5298bfe250a3b661869a26f67a94c231afc1cf9c fix(stm32mp2): syscfg: Add delay block stop API]
Change-Id: Ia15da583ed9bdb92e822aeaa4eb1e48bfeaa1c4b
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/316824
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/include/stm32mp2_private.h | 1 +
 plat/st/stm32mp2/stm32mp2_syscfg.c          | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/plat/st/stm32mp2/include/stm32mp2_private.h b/plat/st/stm32mp2/include/stm32mp2_private.h
index be1fe825d4..790479a516 100644
--- a/plat/st/stm32mp2/include/stm32mp2_private.h
+++ b/plat/st/stm32mp2/include/stm32mp2_private.h
@@ -19,6 +19,7 @@ int stm32mp2_syscfg_dlyb_init(uint8_t bank, bool bypass_mode,
 			      uint16_t period_ps);
 int stm32mp2_syscfg_dlyb_find_tap(uint8_t bank, int (*check_transfer)(void),
 				  bool rx_only);
+void stm32mp2_syscfg_dlyb_stop(uint8_t bank);
 
 /* IO compensation identifiers */
 enum syscfg_io_ids {
diff --git a/plat/st/stm32mp2/stm32mp2_syscfg.c b/plat/st/stm32mp2/stm32mp2_syscfg.c
index b4ec10c97d..497fb6b35a 100644
--- a/plat/st/stm32mp2/stm32mp2_syscfg.c
+++ b/plat/st/stm32mp2/stm32mp2_syscfg.c
@@ -221,6 +221,11 @@ static int stm32_syscfg_dlyb_set_tap(uint8_t bank, uint8_t tap, bool rx_tap)
 	return 0;
 }
 
+void stm32mp2_syscfg_dlyb_stop(uint8_t bank)
+{
+	mmio_write_32(SYSCFG_BASE + syscfg_dlybos_cr_offset[bank], 0x0);
+}
+
 int stm32mp2_syscfg_dlyb_find_tap(uint8_t bank, int (*check_transfer)(void),
 				  bool rx_only)
 {
-- 
2.53.0

